src/storage/txn/txn_status_cache.rs

```diff
@@ -1513,4 +1522,53 @@ mod tests {
         assert!(removed.is_some());
         assert_eq!(cache.get(1.into()), None);
     }
+
+    #[test]
+    fn test_txn_status_cache_lock_order() {
+        use std::{sync::Arc, thread, time::Duration};
+
+        use fail::FailScenario;
+        use txn_types::TimeStamp;
+
+        let scenario = FailScenario::setup();
+        fail::cfg("txn_status_cache_get_after_large_before_normal", "pause").unwrap();
+        fail::cfg("txn_status_cache_upsert_after_normal_before_large", "pause").unwrap();
+
+        let cache = Arc::new(TxnStatusCache::with_slots_and_time_limit(
+            1, // Use single slot for maximum contention
+            Duration::from_secs(30),
+            Duration::from_secs(60),
+            1000,
+        ));
+
+        let start_ts = TimeStamp::from(42);
+        let now = SystemTime::now();
+
+        // Start thread 1 (upsert operation)
+        let cache_clone1 = cache.clone();
+        let thread1 = thread::spawn(move || {
+            let state = TxnState::Committed {
+                commit_ts: TimeStamp::from(43),
+            };
+            cache_clone1.upsert(start_ts, state, now);
+        });
+
+        // Start thread 2 (get operation)
+        let cache_clone2 = cache.clone();
+        let thread2 = thread::spawn(move || {
+            let _ = cache_clone2.get(start_ts);
+        });
+
+        // Wait a moment to ensure both threads have reached their failpoints
+        thread::sleep(Duration::from_millis(300));
+
+        // Release the failpoints
+        fail::remove("txn_status_cache_get_after_large_before_normal");
+        fail::remove("txn_status_cache_upsert_after_normal_before_large");
+
+        // if deadlock, thread1 will not join, test will hang here
+        thread1.join().unwrap();
+        thread2.join().unwrap();
+        drop(scenario);
```

Comment:
Is this needed?

Reply:
Not necessary, but just to make the usage explcit. Removed